analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -71,121 +71,227 @@
     bicycle_parking.id IS NULL
 """
 
+
 sql30 = """
-CREATE TEMP TABLE oneway AS
+-- Collect all oneways
+CREATE TEMP TABLE oneway_highway AS
 SELECT
   id,
-  t.nid,
+  nodes,
+  tags?'oneway' AND tags->'oneway' = '-1' AS oneway_reverse,
+  linestring,
+  array_length(nodes, 1) AS length,
+  generate_subscripts(nodes, 1) AS nid_index
+FROM
+  highways
+WHERE
+  (is_oneway OR is_roundabout) AND
+  NOT is_area AND
+  NOT is_construction
+"""
+
+sql31 = """
+CREATE INDEX idx_oneway_highway_linestring ON oneway_highway USING GIST(linestring);
+"""
+
+sql32 = """
+-- Collect all nodes that could be possible (valid) end nodes
+CREATE TEMP TABLE allowed_end_nodes AS
+SELECT DISTINCT
+  unnest(nodes) AS id
+FROM
+  ways
+WHERE
+  ways.tags != '' AND
+  (
+    (ways.tags?'amenity' AND ways.tags->'amenity' = 'parking') OR
+    (ways.tags?'railway' AND ways.tags->'railway' = 'platform') OR
+    (ways.tags?'aeroway' AND ways.tags->'aeroway' = 'taxiway')
+  )
+UNION ALL
+SELECT
+  id
+FROM
+  nodes
+WHERE
+  tags != '' AND
+  (
+    (tags?'amenity'  AND tags->'amenity' = 'parking_entrance') OR
+    (tags?'entrance' AND tags->'entrance'!= 'garage')
+  )
+UNION ALL
+SELECT DISTINCT
+  unnest(nodes) AS id
+FROM
+  highways
+WHERE
+  -- Non-oneway highways are valid input nodes for oneways
+  (NOT is_oneway AND NOT is_roundabout) OR
+  -- Force motorways as valid input nodes
+  -- Raceways are commonly isolated from the main network
+  -- Escape (emergency stop) ways are supposed to be not used / dead-ended
+  -- Footway can lead to anything (attraction entrances, ...) so exclude them even if oneway
+  highway IN ('motorway', 'raceway', 'escape', 'footway') OR
+  -- Construction roads are usually temporary, the connections are likely access=no or similar
+  is_construction
+UNION ALL
+-- Include nodes outside the extract; the recursive check will otherwise fail on oneways originating outside of the extract. See #1949
+SELECT DISTINCT
+  way_nodes.node_id
+FROM
+  way_nodes
+  LEFT JOIN nodes AS n ON
+    n.id = way_nodes.node_id
+WHERE
+  n IS NULL
```

Comment:
Look expensive to me, doesn't ? Maybe there is other ways to do this.

Reply:
It is the most expensive part indeed (if you exclude sql20, which I left untouched). Do you think it would be better to:
1. Get all ways that cross an extract boundary (like we do here: https://github.com/osm-fr/osmose-backend/blob/dev/analysers/analyser_osmosis_highway_floating_islands.py#L48-L60 )
2. Then 'unnest' all nodes of that way into this table? 

Note: we can't check if the nodes are NULL then: it may cross the boundary and directly after that become a different way (still in the extract, but outside the boundary), which can have the NULL nodes. Hence we'd have to grab all nodes of that boundary-crossing highway

Reply:
The 1 Look good to me. Way fully outside of the boundary will be clipped anyway by later result sanitation.

Reply:
I'll have a look later this week!

Reply:
Done! It's about a factor 2.5 faster on larger extracts (like japan_chubu)